tests/rptest/tests/availability_test.py

```diff
@@ -104,3 +104,5 @@ def test_recovery_after_catastrophic_failure(self):
         self.inject_failure(f_spec)
 
         self.validate_records()
+
+        self._continue_all()
```

Comment:
What if there is a test failure? Also, looks like there is a race condition between this and `_failure_injector_loop` running in a separate thread...

I see that this `EndToEndFinjectorTest` is not used much, maybe it should be rewritten from a test base class to a context manager, so that we can write `with finjector_thread() as _:` in the tests.

Reply:
> What if there is a test failure?

It'll throw one more when gave up trying to stop Redpanda. How bad is it?

> Also, looks like there is a race condition between this and _failure_injector_loop running in a separate thread

Not really. First, it only applies to `test_availability_when_one_node_failed` where we launch the failures thread. But that test only injects temporary suspensions for up to 20 seconds, and the stop timeout is 30 seconds.

Reply:
I pushed a version with context managers, please could you have another look?